docs/api-guide.md

```diff
@@ -786,107 +786,115 @@ Let's look at the following example code to see how to handle exceptions in Scal
-          return;
+          // Throw the last exception if the number of retries exceeds the maximum
+          throw lastException;
         }
+
         // Sleep 100 milliseconds before retrying the transaction in this sample code
         TimeUnit.MILLISECONDS.sleep(100);
       }
 
-      // Begin a transaction
-      DistributedTransaction tx;
+      DistributedTransaction transaction = null;
       try {
-        tx = transactionManager.begin();
-      } catch (TransactionNotFoundException e) {
-        // if the transaction fails to begin due to transient faults. You can retry the transaction
-        continue;
-      } catch (TransactionException e) {
-        // If beginning a transaction failed, it indicates some failure happens during the
-        // transaction, so you should cancel the transaction or retry the transaction after the
-        // failure or error is fixed
-        return;
-      }
+        // Begin a transaction
+        transaction = transactionManager.begin();
 
-      try {
         // Execute CRUD operations in the transaction
-        Optional<Result> result = tx.get(...);
-        List<Result> results = tx.scan(...);
-        tx.put(...);
-        tx.delete(...);
+        Optional<Result> result = transaction.get(...);
+        List<Result> results = transaction.scan(...);
+        transaction.put(...);
+        transaction.delete(...);
 
         // Commit the transaction
-        tx.commit();
-      } catch (CrudConflictException | CommitConflictException e) {
-        // If you catch CrudConflictException or CommitConflictException, it indicates a transaction
-        // conflict occurs during the transaction, so you can retry the transaction from the 
-        // beginning
-        try {
-          tx.rollback();
-        } catch (RollbackException ex) {
-          // Rolling back the transaction failed. You can log it here
-        }
+        transaction.commit();
       } catch (UnsatisfiedConditionException e) {
-        // You need to handle UnsatisfiedConditionException only if a mutation operation specifies a condition.
-        // This exception indicates the condition for the mutation operation is not met, so you can
-        // retry the transaction once the exception cause is fixed
-        try {
-          tx.rollback();
-        } catch (RollbackException ex) {
-          // Rolling back the transaction failed. You can log it here
-        }
-      } catch (CrudException | CommitException e) {
-        // If you catch CrudException or CommitException, it indicates some failure happens, so you
-        // should cancel the transaction or retry the transaction after the failure or error is 
-        // fixed
+        // You need to handle `UnsatisfiedConditionException` only if a mutation operation specifies
+        // a condition. This exception indicates the condition for the mutation operation is not met
+
         try {
-          tx.rollback();
+          transaction.rollback();
```

Comment:
[trivial] I guess those who read the error handling for `rollback` might wonder why just logging is okay. Maybe describing the reason that the failures of `rollback` can be ignored would be helpful for users.

Reply:
Thank you. I added some more explanation in ad4828e21cf86f30b1eae84b30aed1c8eb0653d8. Please take a look when you have time!